src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java

```diff
@@ -58,10 +59,12 @@ private XdocUtil() {
      */
     public static Set<Path> getXdocsFilePaths() throws IOException {
         final Path directory = Paths.get(DIRECTORY_PATH);
-        try (DirectoryStream<Path> stream = Files.newDirectoryStream(directory, "*.xml")) {
+        try (Stream<Path> stream = Files.walk(directory)) {
             final Set<Path> xdocs = new HashSet<>();
-            for (Path entry : stream) {
-                xdocs.add(entry);
+            for (Path path : stream.collect(Collectors.toList())) {
+                if (Files.isRegularFile(path) && path.toString().endsWith(".xml")) {
+                    xdocs.add(path);
+                }
```

Comment:
It is probably better to do (psuedocode):
```
Set xdocs = Files.find(path, (f,a) -> a.isRegularFile && f.path.toString.endsWith(".xml")).collect(Collectors.toSet));

```
Filtering by predicate should improve performance.

Reply:
Done.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#find-java.nio.file.Path-int-java.util.function.BiPredicate-java.nio.file.FileVisitOption...-